tiny_cnn/lossfunctions/loss_function.h

```diff
@@ -32,48 +32,67 @@ namespace tiny_cnn {
 // mean-squared-error loss function for regression
 class mse {
 public:
-    static float_t f(float_t y, float_t t) {
-        return (y - t) * (y - t) / 2;
+    static float_t f(const vec_t& y, const vec_t& t) {
+        assert(y.size() == t.size());
+        float_t d = 0.0;
+        for(unsigned int i = 0; i < y.size(); ++i)
+            d += (y[i] - t[i]) * (y[i] - t[i]);
+        return d/y.size();
     }
 
```

Comment:
In #217 I mean that we should keep a coherence between definitions of ```f``` and ```df```. If you change the definition of ```f``` to ```1/N * sum(y[i] - t[i])^2```,  ```df``` must be ```2/N * sum(y[i] - t[i])```.